packages/components/src/higher-order/with-focus-return/index.js

```diff
@@ -69,14 +69,9 @@ function withFocusReturn( options ) {
 			componentWillUnmount() {
 				const {
 					activeElementOnMount,
-					isFocused,
 					ownFocusedElements,
 				} = this;
 
-				if ( ! isFocused ) {
-					return;
-				}
-
```

Comment:
It'd be interesting to know the reasoning behind removing this code. I'm not sure what it does, but I think it terms of documenting its removal some details in the PR description would be great.

Reply:
Yes, this seems dangerous for me. This means if we close a modal/popover programmatically (not by focusing something else explicitly) and the focus is outside of that modal/popover when we do so, the focus will return to the button that opened the popover) which IMO is not what we want right?

Reply:
> I'm not sure what it does, but I think it terms of documenting its removal some details in the PR description would be great.

What it currently does is return if no element within the wrapped component is focused, so that focus goes back to the document body. I can't think of any situation where we would want this to happen, especially when using a component the sole purpose of which is to return focus to the previously focused element when closing a modal or popover. So I'm particularly curious to know why it was added in the first place. I know it's part of [this changeset](https://github.com/WordPress/gutenberg/pull/14444) but I can't reproduce the issue it fixes after my changes.

Note that this return doesn't get triggered when we deliberately move focus to a specific element, such as when we use the Block Navigator. It only gets triggered when focus is for some reason lost while we are still inside the wrapped component. This is something that, for the sake of accessibility, should never happen, so I'm not sure that we should be catering to it.

Removing this code fixes the issue of focus being lost when closing the font size dropdown because in the `focusActiveItem` function [focus is blurred from the dropdown just before closing](https://github.com/WordPress/gutenberg/pull/17418/files#diff-85dcbe8ad832b71a98885ebcbf5695d9R45). I'm not sure why we're managing focus with this function though; I thought `NavigableMenu` would be enough to provide keyboard navigation with the arrow keys? Perhaps @youknowriad could shed some light on this. 

Happy to hear further feedback and suggestions!

Reply:
I think this check exists to address the issues that there might be several nested components that are wrapped with `withFocusReturn` and all of them can respond when they get unmounted. I suspect, this is a way to ensure that it's handled only once in such a case. This should be confirmed though.

Reply:
> I think this check exists to address the issues that there might be several nested components that are wrapped with withFocusReturn and all of them can respond when they get unmounted.

Hmmm I don't think that's what it does, because if we have nested `withFocusReturn`s (such as this font size picker nested in the sidebar) the parent keeps tracking focus even when it's inside the child component, so I wouldn't expect focus to be lost in the parent unless it were also lost in the child.

We don't seem to have any instance of a parent popover/withFocusReturn-wrapped component that closes when a child popover is closed (that's probably an accessibility anti-pattern anyway), but I have verified that, with the current changes, when opening the font size picker and then closing the sidebar with it still open, focus is correctly transferred to the open sidebar button.